src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -280,15 +310,74 @@ private boolean parseNodes(List<Node> nodes) {
           continue;
         }
 
-        if (cpeUriCheck(cpeMatch)) {
+        if (projectCheck(cpeMatch, referenceMatch, project)) {
           return true;
         }
       }
 
-      if (parseNodes(node.getChildren())) {
+      if (parseNodes(node.getChildren(), referenceMatch, project)) {
         return true;
       }
     }
     return false;
   }
+
+  /**
+   * Checks if the input {@link URI} is same as the project's URL.
+   * 
+   * @param referenceUrl of type {@link URI}.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the input URL matches the project's URL. Otherwise false.
+   */
+  private static boolean checkUrl(URI referenceUrl, GitHubProject project) {
+    if (referenceUrl.getHost().equals(project.url().getHost())
+        && checkUrlPath(referenceUrl.getPath(), project)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Extract {@link URI} from the input string URL.
+   * 
+   * @param url Input string URL.
+   * @return type {@link URI} or an {@link Optional#empty()}.
+   */
+  private static Optional<URI> getUrl(String url) {
+    try {
+      return Optional.ofNullable(new URI(url));
+    } catch (URISyntaxException e) {
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Matches the reference URL with the project's URL. The URL must not contain the pre-defined stop
+   * words.
+   * 
+   * @param path The path of the URL.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the project URL matches with the reference URL. Otherwise False.
+   */
+  private static boolean checkUrlPath(String path, GitHubProject project) {
+    String delimiter = "/";
+    String[] splitPath = StringUtils.stripEnd(path, delimiter).split(delimiter);
```

Comment:
As I mentioned earlier, the method still expects that the path starts with `/`. If you are sure it's safe, we can keep it as-is. But I would remove the leading '/' (not trailing ones) and them used `splitPath[0]` and `splitPath[1]`.

Reply:
Now I understand. I will do that.

Reply:
The `URI.getPath()`. I think does not send a leading `/`.